airflow/providers/mongo/hooks/mongo.py

```diff
@@ -98,12 +101,13 @@ def get_conn(self) -> MongoClient:
 
         # If we are using SSL disable requiring certs from specific hostname
         if options.get("ssl", False):
+            allow_insecure = self.extras.get("allow_insecure", True)
             if pymongo.__version__ >= "4.0.0":
-                # In pymongo 4.0.0+ `tlsAllowInvalidCertificates=True`
-                # replaces `ssl_cert_reqs=CERT_NONE`
-                options.update({"tlsAllowInvalidCertificates": True})
+                # For pymongo 4.0.0+, use 'tlsAllowInvalidCertificates'
```

Comment:
Pymongo 4.0 was released in 2021 maybe we should just support >=4.0?

Reply:
Yep makes sense